From 9180f96007eb91fe22c1b572bf7c83d866ac52dc Mon Sep 17 00:00:00 2001
From: Praveen Talari <praveen.talari@oss.qualcomm.com>
Date: Fri, 13 Mar 2026 10:34:35 +0530
Subject: [PATCH] PENDING: spi: geni-msm: Reconfigure SPI mode after resume

Initialize last_mode and last_cs to some default value (0 & 0xFF) during
system resume to ensure proper hardware reconfiguration.

Setting last_mode and last_cs to 0 and 0xFF guarantees that any valid mode
and CS request after resume is treated as a change and triggers the
necessary update.

Change-Id: Ic89673d16f9c562fa99f45251298961268365387
Signed-off-by: Praveen Talari <praveen.talari@oss.qualcomm.com>
---
 drivers/spi/spi-geni-qcom.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/spi/spi-geni-qcom.c b/drivers/spi/spi-geni-qcom.c
index 97f8b5aafdfa..9ce16bf1908c 100644
--- a/drivers/spi/spi-geni-qcom.c
+++ b/drivers/spi/spi-geni-qcom.c
@@ -1412,7 +1412,13 @@ static int __maybe_unused spi_geni_resume(struct device *dev)
 	if (pm_suspend_target_state == PM_SUSPEND_MEM) {
 		dev_dbg(dev, "%s: In S2R\n", __func__);
 		mas->cur_xfer_mode = GENI_SE_INVALID;
+
+		/*
+		 * Initialize last_mode and last_cs to 0 and 0xFF during system resume
+		 * to force hardware reconfiguration after DSQB.
+		 */
 		mas->last_mode = 0;
+		mas->last_cs = 0xFF;
 		ret = spi_geni_init(mas);
 		if (ret)
 			return ret;
-- 
2.34.1

